android/src/app/organicmaps/routing/SingleLaneInfo.java

```diff
@@ -1,13 +1,49 @@
 package app.organicmaps.routing;
 
+import androidx.annotation.DrawableRes;
+import app.organicmaps.R;
+
 public class SingleLaneInfo
 {
-  byte[] mLane;
+  LaneWay[] mLane;
   boolean mIsActive;
 
-  SingleLaneInfo(byte[] lane, boolean isActive)
+  /**
+   * IMPORTANT : Order of enum values MUST BE the same
+   * with native LaneWay enum (see routing/turns.hpp for details).
+   * Information for every lane is composed of some number values below.
+   * For example, a lane may have THROUGH and RIGHT values.
+   */
+  public enum LaneWay
   {
-    mLane = lane;
+    NONE(R.drawable.ic_turn_straight),
+    REVERSE(R.drawable.ic_turn_uleft),
```

Comment:
Interesting here. Should we distinguish U-Turn-left, U-Turn-right?

Reply:
It would be a little nicer. However, I expect users will understand what it means without a left/right variant.

The tagging doesn't distinguish left vs right, so we would have to guess from the side of the road or the direction of traffic in the country.

On taginfo, `reverse` is used in fewer than 1% of `turn:lanes`, `turn:lanes:forward`, and `turn:lanes:backward` tags, so it isn't something that will come up often.

For those reasons, I decided against adding support for it, at least in this PR. If this is an important feature, I can work on opening another PR to add that support.

Reply:
Agree.

Reply:
Let's create a separate detailed issue for it.

Reply:
Just opened #4810 for this